src/Symfony/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Symfony\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder('async_aws');
+        /** @var ArrayNodeDefinition $rootNode */
+        $rootNode = $treeBuilder->getRootNode();
+        $node = $rootNode->children();
+        $node->append($this->addClientSection('default'));
+
+        foreach (AwsPackagesProvider::getAllServices() as $key => $data) {
+            $node->append($this->addClientSection($key));
+        }
+        $node->end();
+
+        return $treeBuilder;
+    }
+
+    private function addClientSection(string $name): ArrayNodeDefinition
+    {
+        $treeBuilder = new TreeBuilder($name);
+        /** @var ArrayNodeDefinition $rootNode */
+        $rootNode = $treeBuilder->getRootNode();
+
+        $node = $rootNode->children();
+
+        if ('default' !== $name) {
+            $node->booleanNode('register_service')->info('If set to false, no service will be created.')->defaultTrue()->end();
+        }
+
+        $node
+            ->arrayNode('config')->arrayPrototype()->end()
+            ->scalarNode('credential_provider')->info('A service name for AsyncAws\Core\Credentials\CredentialProvider.')->defaultNull()->end()
```

Comment:
I'm not sure we need such configuration. Couldn't they decorates the bundle's internal services instead?

Reply:
Na. 
Currently the library decides what should happen if nothing is configured.